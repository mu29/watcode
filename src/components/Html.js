/* eslint-disable react/no-danger */
import React from 'react'
import { PropTypes } from 'prop-types'
import { Helmet } from 'react-helmet'

const Html = ({ styles, state, children }) => {
  const helmet = Helmet.renderStatic()
  const htmlAttrs = helmet.htmlAttributes.toComponent()
  const bodyAttrs = helmet.bodyAttributes.toComponent()

  return (
    <html lang="ko" { ...htmlAttrs }>
      <head>
        { helmet.title.toComponent() }
        { helmet.meta.toComponent() }
        { helmet.link.toComponent() }
        <meta charSet="utf8" />
        <link href="/style.css" rel="stylesheet" type="text/css" />
        <link href="//spoqa.github.io/spoqa-han-sans/css/SpoqaHanSans-kr.css" rel="stylesheet" type="text/css" />
        <script defer src="https://use.fontawesome.com/releases/v5.0.13/js/all.js" integrity="sha384-xymdQtn1n3lH2wcu0qhcdaOpQwyoarkgLVxC/wZ5q7h9gHtxICrpcaSUfygqZGOe" crossOrigin="anonymous" />
        { styles }
      </head>
      <body { ...bodyAttrs }>
        <main id="app">{ children }</main>
        <script dangerouslySetInnerHTML={{ __html: state }} />
        <script src="/client.js" />
      </body>
    </html>
  )
}

Html.propTypes = {
  styles: PropTypes.node.isRequired,
  state: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
}

export default Html
