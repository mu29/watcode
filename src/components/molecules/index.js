export { default as Field } from './Field'
export { default as IconButton } from './IconButton'
export { default as Tab } from './Tab'
