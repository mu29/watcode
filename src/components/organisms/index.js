export { default as ArtworkList } from './ArtworkList'
export { default as Header } from './Header'
export { default as LoginForm } from './LoginForm'
export { default as ReduxField } from './ReduxField'
export { default as RegisterForm } from './RegisterForm'
export { default as SearchBar } from './SearchBar'
