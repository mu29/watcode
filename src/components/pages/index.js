export { default as HomePage } from './HomePage'
export { default as RankingPage } from './RankingPage'
export { default as BookmarkPage } from './BookmarkPage'
