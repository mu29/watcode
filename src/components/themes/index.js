export { default as basicTheme } from './basic'
