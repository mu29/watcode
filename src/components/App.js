import React from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import { injectGlobal, ThemeProvider } from 'styled-components'
import { authorizeActions, readBookmarksActions } from 'store/actions'
import { auth } from 'services/firebase'
import { basicTheme } from './themes'
import Router from './Router'

/* eslint-disable no-unused-expressions */
injectGlobal`
  * {
    box-sizing: border-box;
  }
  body {
    margin: 0;
  }
`

export default class App extends React.Component {
  static propTypes = {
    store: PropTypes.shape({
      dispatch: PropTypes.func,
      getState: PropTypes.func,
    }).isRequired,
  }

  componentWillMount() {
    const { store } = this.props
    auth.onAuthStateChanged((user) => {
      store.dispatch(authorizeActions.request({ user }))
      store.dispatch(readBookmarksActions.request())
    })
  }

  render() {
    const state = this.props.store.getState()
    const isLoggedIn = state.auth.user.email !== undefined
    return (
      <React.Fragment>
        <Helmet titleTemplate="watcode">
          <title>왓코드 - 오늘 볼 만화 추천해줘!</title>
          <meta charSet="UTF-8" />
          <meta name="description" content="" />
          <meta name="viewport" content="width=device-width, initial-scale=1.0" />
          <meta property="og:site_name" content="watcode" />
        </Helmet>
        <ThemeProvider theme={ basicTheme }>
          <Router isLoggedIn={ isLoggedIn } />
        </ThemeProvider>
      </React.Fragment>
    )
  }
}
