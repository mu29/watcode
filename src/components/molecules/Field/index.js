import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { Label, Input } from 'components'
import { palette } from 'services/style'

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`

const LabelArea = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 0.25rem;
  padding: 0.25rem;
`

const Error = styled.span`
  display: flex;
  align-items: center;
  color: ${palette('red.default')};
  font-size: 0.75rem;
`

const Field = ({
  type, name, label, error, ...props
}) => {
  const inputProps = {
    id: name,
    name,
    type,
    ...props,
  }

  return (
    <Wrapper error={ error }>
      {
        (label || error) &&
        <LabelArea>
          { label && <Label htmlFor={ inputProps.id }>{ label }</Label> }
          { error && <Error>{ error }</Error> }
        </LabelArea>
      }
      <Input { ...inputProps } error={ error } />
    </Wrapper>
  )
}

Field.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string,
  label: PropTypes.string,
  error: PropTypes.string,
  touched: PropTypes.bool,
  className: PropTypes.string,
}

Field.defaultProps = {
  type: 'text',
  label: undefined,
  error: undefined,
  touched: false,
  className: undefined,
}

export default Field
