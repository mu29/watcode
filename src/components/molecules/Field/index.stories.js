import React from 'react'
import { storiesOf } from '@storybook/react'
import Field from '.'

storiesOf('Molecules/Field', module)
  .add('기본', () => (
    <Field name="field" placeholder="메일 주소를 입력해주세요" />
  ))
  .add('라벨', () => (
    <Field name="field" label="이메일" placeholder="메일 주소를 입력해주세요" />
  ))
  .add('오류', () => (
    <Field name="field" label="이메일" placeholder="메일 주소를 입력해주세요" error="존재하지 않는 계정입니다." />
  ))
