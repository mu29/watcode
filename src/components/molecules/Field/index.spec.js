import React from 'react'
import { shallow } from 'enzyme'
import { basicTheme } from 'components/themes'
import 'jest-styled-components'
import Field from '.'

const wrap = (props = {}) => shallow(<Field theme={ basicTheme } name="name" { ...props } />).dive()

describe('<Field />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = wrap()
    expect(wrapper).toMatchSnapshot()
  })

  it('name 속성을 <Input /> id로 넘겨준다', () => {
    const wrapper = wrap()
    expect(wrapper.find({ id: 'name' })).toHaveLength(1)
  })

  it('label 속성이 들어오면 그려준다', () => {
    const wrapper = wrap({ label: 'foo label' })
    expect(wrapper).toMatchSnapshot()
  })

  it('오류 메시지를 표시한다', () => {
    const wrapper = wrap({ error: 'foo error' })
    expect(wrapper).toMatchSnapshot()
  })
})
