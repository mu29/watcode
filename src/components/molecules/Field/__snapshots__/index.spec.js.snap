// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> label 속성이 들어오면 그려준다 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<div
  className="c0"
>
  <LabelArea>
    <Label
      color="gray.90"
      font="default"
      fontSize={12}
      htmlFor="name"
    >
      foo label
    </Label>
  </LabelArea>
  <Input
    block={false}
    border="gray.30"
    color="gray.90"
    font="default"
    fontSize={16}
    id="name"
    name="name"
    primaryColor="yellow.default"
    theme={
      Object {
        "fonts": Object {
          "default": "'Spoqa Han Sans', 'Sans-serif', 'Helvetica Neue', 'Nanum Gothic'",
          "monospace": "'Monaco', 'Consolas'",
        },
        "palette": Object {
          "gray": Object {
            "0": "#FAFAFA",
            "10": "#F5F5F5",
            "100": "#212121",
            "20": "#EEEEEE",
            "30": "#E0E0E0",
            "40": "#BDBDBD",
            "60": "#9E9E9E",
            "70": "#757575",
            "80": "#616161",
            "90": "#424242",
          },
          "red": Object {
            "default": "#FF8787",
          },
          "transparent": Object {
            "default": "transparent",
          },
          "white": Object {
            "default": "#FFF",
          },
          "yellow": Object {
            "default": "#F9D14C",
          },
        },
        "sizes": Object {},
      }
    }
    touched={false}
    type="text"
  />
</div>
`;

exports[`<Field /> 오류 메시지를 표시한다 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<div
  className="c0"
>
  <LabelArea>
    <Error>
      foo error
    </Error>
  </LabelArea>
  <Input
    block={false}
    border="gray.30"
    color="gray.90"
    error="foo error"
    font="default"
    fontSize={16}
    id="name"
    name="name"
    primaryColor="yellow.default"
    theme={
      Object {
        "fonts": Object {
          "default": "'Spoqa Han Sans', 'Sans-serif', 'Helvetica Neue', 'Nanum Gothic'",
          "monospace": "'Monaco', 'Consolas'",
        },
        "palette": Object {
          "gray": Object {
            "0": "#FAFAFA",
            "10": "#F5F5F5",
            "100": "#212121",
            "20": "#EEEEEE",
            "30": "#E0E0E0",
            "40": "#BDBDBD",
            "60": "#9E9E9E",
            "70": "#757575",
            "80": "#616161",
            "90": "#424242",
          },
          "red": Object {
            "default": "#FF8787",
          },
          "transparent": Object {
            "default": "transparent",
          },
          "white": Object {
            "default": "#FFF",
          },
          "yellow": Object {
            "default": "#F9D14C",
          },
        },
        "sizes": Object {},
      }
    }
    touched={false}
    type="text"
  />
</div>
`;

exports[`<Field /> 정상적으로 렌더링된다 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

<div
  className="c0"
>
  <Input
    block={false}
    border="gray.30"
    color="gray.90"
    font="default"
    fontSize={16}
    id="name"
    name="name"
    primaryColor="yellow.default"
    theme={
      Object {
        "fonts": Object {
          "default": "'Spoqa Han Sans', 'Sans-serif', 'Helvetica Neue', 'Nanum Gothic'",
          "monospace": "'Monaco', 'Consolas'",
        },
        "palette": Object {
          "gray": Object {
            "0": "#FAFAFA",
            "10": "#F5F5F5",
            "100": "#212121",
            "20": "#EEEEEE",
            "30": "#E0E0E0",
            "40": "#BDBDBD",
            "60": "#9E9E9E",
            "70": "#757575",
            "80": "#616161",
            "90": "#424242",
          },
          "red": Object {
            "default": "#FF8787",
          },
          "transparent": Object {
            "default": "transparent",
          },
          "white": Object {
            "default": "#FFF",
          },
          "yellow": Object {
            "default": "#F9D14C",
          },
        },
        "sizes": Object {},
      }
    }
    touched={false}
    type="text"
  />
</div>
`;
